packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -29,6 +29,18 @@ enum FlexDirection {
   vertical
 }
 
+/// Which incoming constraint parameter defines how much free space is
+/// available along the main axis in a flex layout.
+enum MainAxisSpace {
+  /// The free space is the difference between incoming constraint's min parameter
+  /// and the collective size of the children, but not less than zero.
+  min,
+
+  /// The free space is the difference between incoming constraint's max parameter
+  /// and the collective size of the children, but not less than zero.
```

Comment:
I've renamed everything mainAxisSpace to mainAxisSize.

What I'd meant to get at by pointing out the freeSpace is >=0 is that we don't allocate negative free space.  If the freeSpace is effectively < 0 we wouldn't make elements overlap etc.